### DISCUSSION TITLE
How to connect nos3 with fprime？

### DISCUSSION THREAD

**Discussion by WiilyLin** [nasa/nos3#211] [General]
Well, I try to connect the nos3 with fprime. Because I already make the fprime model. And I need simulator in NOS3. Now I just know maybe I need to replace the cFe with fprime. I found the author wrote some suggestion in https://github.com/nasa/nos3/wiki/NOS3-cFS-Development. But I also not sure the connection between the cFe and NOS3. What is I/O for cFe. Where the NOS3 receive from FSW?
Are there any suggestion to find them?

Thanks~

**Comment by jlucas9** [comment]
The [hardware library](https://github.com/nasa/nos3/wiki/NOS3-cFS-Development#hardware-library) (nos3/fsw/apps/hwlib) is what enables the communication between flight software and simulators. This library would need to be included in your F' build and used by your FSW application in order to enable communications. Note you would also need to add hooks into F' to enable NOS3 to control the time tics used in order to keep FSW and Simulators in step. cFS enables this by providing a Platform Support Package (PSP) and Operating System Abstraction Layer (OSAL) where we can add in the required hooks. Unfortunately, I haven't dug into F' yet so I don't know if it has similar capabilities or where specifically these would go in the code.

**Comment by jlucas9** [comment]
If you're trying to build your own base box or install NOS3 on your host machine (Linux based) you can leverage the deployment repository at: https://github.com/nasa-itc/deployment 

The above includes a `deploy-local-env.sh` script that may need edited some to get you deployed. Each deployment has it's own difficulties which is why we've moved to provide a base box for others to immediately download and use.

**Comment by jlucas9** [comment]
Note that an issue to further track F' integration exists - https://github.com/nasa/nos3/issues/84
